Src/DependencyCollector/NetCore/DependencyCollectorDiagnosticListener.cs

```diff
@@ -126,7 +124,7 @@ public void OnActivityStart(HttpRequestMessage request)
             if (request != null &&
                 !this.applicationInsightsUrlFilter.IsApplicationInsightsUrl(request.RequestUri.ToString()))
             {
-                this.InjectRequestHeaders(request, this.client.Context.InstrumentationKey);
+                this.InjectRequestHeaders(request, this.configuration.InstrumentationKey);
```

Comment:
so this will not support the `roleName` integration that @nizarq was working on? I'd suggest to use the same approach he used - take instrumentation key from the `DependencyTelemetry` object. It is already initializaed so it will have it